### PR TITLE
Refresh mgmt signal logic to remove possible memory issues.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
~This is an effort to remove potential causes for problems with #4413.~

I was hoping this cleanup would let me either remove or fix `LLQ` but that's embedded in unpleasant ways in other code. It's only my KIWF list, though. Someday...

### CODE REVIEW

**Comment by chitianhao** [comment]
This is nice.

File: mgmt/BaseManager.cc [hunk 1]
```diff
@@ -22,35 +22,51 @@
...
+  while (!queue.empty()) {
+    ats_free(queue.front());
+    queue.pop();
   }
-  ats_free(mgmt_event_queue);
+}
 
-  for (auto &&it : mgmt_callback_table) {
-    MgmtCallbackList *tmp, *cb_list = it.second;
+bool
```
Comment (ywkaras) [inline mgmt/BaseManager.cc]: Why isn't this void?

Comment (SolidWallOfCode) [reply mgmt/BaseManager.cc]: I'm not sure. I was following the pattern established in `LLQ`. It should probably be `void` or `self_type`.

File: mgmt/MgmtDefs.h [hunk 2]
```diff
@@ -30,27 +30,29 @@
...
-} MgmtType;
+};
 
 /*
  * MgmtCallback
  *   Management Callback functions.
  */
-typedef void *(*MgmtCallback)(void *opaque_cb_data, char *data_raw, int data_len);
+using MgmtCallback = std::function<void(ts::MemSpan)>;
+// typedef void *(*MgmtCallback)(void *opaque_cb_data, char *data_raw, int data_len);
```
Comment (SolidWallOfCode) [inline mgmt/MgmtDefs.h]: Need to remove this.

File: mgmt/ProcessManager.cc [hunk 3]
```diff
@@ -414,35 +414,35 @@ ProcessManager::handleMgmtMsgFromLM(MgmtMessageHdr *mh)
 {
   ink_assert(mh != nullptr);
 
-  char *data_raw = (char *)mh + sizeof(MgmtMessageHdr);
+  char *payload = static_cast<char *>(mh->payload());
```
Comment (ywkaras) [inline mgmt/ProcessManager.cc]: Should payload() perhaps just return a MemSpan?

File: src/traffic_server/traffic_server.cc [hunk 4]
```diff
@@ -2015,42 +2016,37 @@ REGRESSION_TEST(Hdrs)(RegressionTest *t, int atype, int *pstatus)
...
-  CacheDisk *d = cacheProcessor.find_by_path(arg, len);
-  // Actual command is in @a data.
-  intptr_t cmd = reinterpret_cast<intptr_t>(data);
+  CacheDisk *d = cacheProcessor.find_by_path(arg.data(), int(arg.size()));
 
   if (d) {
     switch (cmd) {
     case MGMT_EVENT_STORAGE_DEVICE_CMD_OFFLINE:
-      Debug("server", "Marking %.*s offline", len, arg);
+      Debug("server", "Marking %.*s offline", int(arg.size()), arg.data());
```
Comment (ywkaras) [inline src/traffic_server/traffic_server.cc]: Using a non-Leif-legal constructor-style cast here.

File: mgmt/ProcessManager.cc (hunk 3 above)
Comment (SolidWallOfCode) [reply mgmt/ProcessManager.cc]: Yeah, that's reasonable.

File: src/traffic_server/traffic_server.cc (hunk 4 above)
Comment (SolidWallOfCode) [reply src/traffic_server/traffic_server.cc]: I'm going to blame you.

**Comment by SolidWallOfCode** [comment]
[approve ci autest]

**Comment by bryancall** [comment]
How is this change related to TCL?  Is one of the goals of this PR to remove the use of ink_llqueue.h?  If so now each use of ink_llqueue.h would need to implement its own thread safe data-structure.  It would be better to create a new thread safe queue that uses stl and toss it into tscore.

**Comment by SolidWallOfCode** [comment]
It's related to TCL because it was derived from work with @chitianhao on #4413. One of the original goals was to remove `ink_llqueue.h` but that turned out to be a larger scope than I originally thought, so this PR was reduced to removing one use of that class and doing some more basic cleanup.

File: mgmt/BaseManager.cc [hunk 5]
```diff
@@ -22,35 +22,50 @@
...
+}
 
 BaseManager::~BaseManager()
 {
-  while (!queue_is_empty(mgmt_event_queue)) {
-    MgmtMessageHdr *mh = (MgmtMessageHdr *)dequeue(mgmt_event_queue);
-    ats_free(mh);
+  while (!queue.empty()) {
+    ats_free(queue.front());
+    queue.pop();
```
[the remark this answers is not in the document]
Comment (SolidWallOfCode) [reply mgmt/BaseManager.cc]: [`std::queue`](http://www.cplusplus.com/reference/queue/queue/) has no method named `pop_front`.

**Comment by SolidWallOfCode** [comment]
Updated to remove references to TCL.

**Pull request merged**